lib/ffi/Cargo.toml

```diff
@@ -14,6 +14,13 @@ vangers = { path = "../.." }
 env_logger = { version = "0.9", optional = true }
 futures = "0.3"
 cgmath = "0.18"
-ron = "0.6"
+ron = "0.5.1"
+
+[target.'cfg(not(target_os = "emscripten"))'.dependencies]
 wgpu = { git = "https://github.com/gfx-rs/wgpu", rev = "9219489", features = ["angle"] }
 hal = { package = "wgpu-hal", git = "https://github.com/gfx-rs/wgpu", rev = "9219489", features = ["gles"] }
+
+[target.'cfg(target_os = "emscripten")'.dependencies]
```

Comment:
this shouldn't be needed at all, since the root Cargo.toml already includes the emscripten and webgl features for `wgpu` and `hal` (unless `ffi` needs them specifically?)

Reply:
yeah, you right, but I don't understand the rule what you used.. so here you have declare features = ["angle"] why it's not done in root Cargo.toml. So I did put this features in root Cargo.toml or ffi Cargo.toml

Reply:
I removed it from root Cargo.toml, and keep only here. I think it's more correct, because root Cargo don't declare hal dependency

Reply:
The way it works is basically that features are added. ffi depends on the main lib, and also depends on hal. So ffi depends on hal by 2 different paths, and whatever features are enabled by any of the paths are added.